uber/site_sections/registration.py

```diff
@@ -774,16 +774,18 @@ def reg_take_report(self, session, **params):
                                                        Attendee.registered > start, Attendee.registered <= end).all()
             params['sales'] = sales
             params['attendees'] = attendees
-            params['total_cash'] = sum(a.amount_paid for a in attendees if a.payment_method == CASH) \
-                                 + sum(s.cash for s in sales if s.payment_method == CASH)
+            params['total_cash'] = sum(a.amount_paid for a in attendees if a.payment_method == c.CASH) \
+                                 + sum(s.cash for s in sales if s.payment_method == c.CASH)
             params['total_credit'] = sum(a.amount_paid for a in attendees if a.payment_method in [c.STRIPE, c.SQUARE, c.MANUAL]) \
                                    + sum(s.cash for s in sales if s.payment_method == c.CREDIT)
         else:
             params['endday'] = localized_now().strftime('%Y-%m-%d')
             params['endhour'] = localized_now().strftime('%H')
             params['endminute'] = localized_now().strftime('%M')
-
-        stations = sorted(filter(bool, Attendee.objects.values_list('reg_station', flat=True).distinct()))
+        # list all reg stations associated with attendees and sales
+        stations_attendees = session.query(Attendee.reg_station).filter(Attendee.reg_station is not None, Attendee.reg_station > 0)
```

Comment:
Common mistake when using SQLAlchemy filters, the `is not None` condition doesn't work how you'd expect it to work. `Attendee.reg_station` is a SQLAclhemy Column property of the `Attendee` class itself, thus it will never be `None`. The condition evaluates to a boolean `True` and is then disregarded by the `.filter()` method.

SQLAclhemy Columns use the overloaded `__eq__` method to generate SQL Conditions. To get this to work the way you want, you have to do `Attendee.reg_station != None` – which evaluates to `attendee.reg_station is not NULL` in SQL.

Reply:
Thanks for the explanation!

Just to clarify, the normal inequality operator is recommended here specifically because the columns are overloading `__eq__`? In normal python code, should one use `is` or `is not` when comparing with None?

Reply:
Correct, normal python code should use `is` or `is not`. This is just a weird quirk of SQLAlchemy that is easy to overlook. I do it by accident all the time, myself!